lib/pacemaker/pcmk_cluster_queries.c

```diff
@@ -274,9 +274,10 @@ ipc_connect(data_t *data, enum pcmk_ipc_server server, pcmk_ipc_callback_t cb)
     }
     rc = pcmk_connect_ipc(api, pcmk_ipc_dispatch_main);
     if (rc != pcmk_rc_ok) {
-        out->err(out, "error: Could not connect to %s: %s",
-                pcmk_ipc_name(api, true),
-                pcmk_rc_str(rc));
+        if (rc != EREMOTEIO) {
+            out->err(out, "error: Could not connect to %s: %s",
+                     pcmk_ipc_name(api, true), pcmk_rc_str(rc));
+        }
```

Comment:
This means that all other callers would have to check for EREMOTEIO and output the error. Maybe a bool argument for whether EREMOTEIO is ok (probably also need to return pcmk_rc_ok if so)

Reply:
My current thought is to return `EREMOTEIO` from `pcmk_cluster_queries.c:ipc_connect()` if `eremoteio_ok (bool)` is true, but without logging an error. That way if the caller needs to do anything special, it can.

In particular, I'd like `pcmk__status()` to output a `pacemakerd-health` message with state "Running on Pacemaker Remote node". To do that, it needs to know that `ipc_connect()` returned `EREMOTEIO`.

Reply:
We could also just skip the `pacemakerd-health` message altogether in that case. We'd just have less detailed info if we fail to connect to the CIB.

Reply:
...yeah, probably better to just skip `pacemakerd-health` for remote nodes.
1. No clean way to do a short vs. friendly state string for "Running on Pacemaker Remote node"
2. From your comment on another thread, it sounds like "Running on" would be misleading since pacemakerd isn't proxied to remote nodes... and maybe we'd EREMOTEIO even if pacemaker_remoted isn't running, I dunno

Reply:
Yep this is a good sign you should test the affected commands on remote nodes (not running, running but not connected, connected)

Reply:
In a later PR (not touching it here), I might rethink this... setting `rc = pcmk_rc_ok` within `ipc_connect()` is fine if and only if we also set `data->pcmkd_state = pcmk_pacemakerd_state_running` when we get `EREMOTEIO`. That no longer seems unreasonable. It seems that `EREMOTEIO` is returned if and only if `pacemaker-remoted` is running (regardless of whether it's connected). That's roughly equivalent to `pacemakerd` running.

The current and planned callers treat `EREMOTEIO` the same as `pcmk_rc_ok && pcmk_pacemakerd_state_running`, so we could cut out the middle man by doing it here if there's no objection.

Reply:
Sounds reasonable